src/test/sampleutiltest.cpp

```diff
@@ -260,32 +261,31 @@ TEST_F(SampleUtilTest, convertS16ToFloat32) {
     for (int i = 0; i < buffers.size(); ++i) {
         CSAMPLE* buffer = buffers[i];
         int size = sizes[i];
-        SAMPLE* s16 = new SAMPLE[size];
+        std::vector<SAMPLE> s16(size);
```

Comment:
I tend to prefer `auto s16 = std::vector<SAMPLE>{size};` in modern C++ code, because it keeps the type and the constructor invocation close to each other. Replace `auto` with `let` and we are close to Rust syntax ;) Ideas or recommendations from your experience?

Just recently I started to prefer curly braces to distinguish initialization from function calls. Not sure if that makes sense as a coding style?

Reply:
I can agree with you, however beware. 
```cpp
// Direct Initialisation via "Aggregate initialization"
auto s16 = std::vector<SAMPLE>{size}; 
```
is very different than
```cpp
 // Direct Initialisation from a "parenthesised list of expressions" or "braced-init-lists"
auto s16 = std::vector<SAMPLE>(size);
```
They are similar, but different, as the "Uniform initialization syntax" is not entirely "uniform". :crying_cat_face:

Reply:
Curly braces work for both cases, i.e. with and without custom constructors, right? It just behaves slightly differently behind the scenes, which should not matter from the callers perspective.

I don't really care. Just use what works and which minimizes redundancy like repeating the type declarations during initialization.

Reply:
> Curly braces work for both cases, i.e. with and without custom constructors, right? 

Nah, sadly this is not the case as the compiler cannot tell what context you mean.

And with "Class template argument deduction" (CTAD), this makes it very difficult.
`std::vector<SAMPLE>(size)` vs `std::vector<SAMPLE>{size}` (or the same in CTAD form, `std::vector{size}`)

I also prefer the style with `auto _ = std::vector{}` when writing in general too. (Rust influences :+1: or from others).